### PR TITLE
[stable/kibana] add Kibana chart

### PR DESCRIPTION
**Pull request by compleatang**
This PR adds a Kibana chart to the stable library.

### CODE REVIEW

File: incubator/kibana/Chart.yaml
Comment (unguiculus) [inline incubator/kibana/Chart.yaml]: Use Github username.

File: incubator/kibana/README.md
Comment (unguiculus) [inline incubator/kibana/README.md]: Remove `(requires Kubernetes >= 1.6)`. We support the current and the previous minor version, which is currently 1.7+.

File: incubator/kibana/templates/deployment.yaml
Comment (unguiculus) [inline incubator/kibana/templates/deployment.yaml]: Only `release` and `app` labels should got here.

File: incubator/kibana/templates/deployment.yaml
Comment (unguiculus) [inline incubator/kibana/templates/deployment.yaml]: `apiVersion: apps/v1beta1`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you explain/document the auth stuff better. You add an ingress for auth but don't add the oauth-proxy or a service for it. Why would that make sense?

**Comment by unguiculus** [comment]
I've just seen #2717. It would probably make sense to add an optional dependency to it once it's merged. Since it has it's own ingress you probably don't need the additional ingress you added here.

**Comment by compleatang** [comment]
Have pushed the requested changes. 

I am of two minds personally when it comes to the optional dependency. On the one hand I can see the handiness of it; on the other hand it doesn't work for my particular use case in that I deploy these under separate meta charts. Also, besides bitly/oauth-proxy there are probably other authentication daemons that are being used within the community. All this said, I'm happy to go in either direction the community thinks best as it is straightforward to add an optional dependency.

As to the extra ingress, the chart is built in line with [this documentation](https://github.com/kubernetes/ingress-nginx/tree/master/docs/examples/external-auth); one important item is that the two ingress objects require different annotations for nginx-ingress to be able to effectively handle the authentication in an automated fashion. This, in turn, requires two different ingress objects. I have not seen (or worked out myself) an ability to collapse both into a single ingress object since we cannot place annotations on particular hosts/paths. If any one has other ideas I am all ears.

**Comment by rendhalver** [comment]
Need help getting this tested?

**Comment by Morriz** [comment]
I assume the oauth-proxy ingress should only live in that oauth-proxy chart, and not here. 
Also, any updates on this? Would love to use a nice clean kibana chart.

**Comment by compleatang** [comment]
Any help moving this forward toward merge status would be welcome from my POV. 

Moving the ingress to another chart is sub-optimal for how we use this chart, namely to protect the domain that kibana is running on.

**Comment by Morriz** [comment]
I believe from an open source chart maintenance perspective it wouldn't be preferred to have other functionality bleed into it. If you want to publish something as open source, please be aware of the unix philosphy. All charts published here are made that way, and do not have any implementation specifics in them.

But your problem touches on big issues with Helm charts in general: how to organize for orchestration of charts, but mostly their inter-part configuration. How do we publish/share sane defaults for likely scenarios?

**Comment by compleatang** [comment]
Agree the "problem" sort of falls in a hole of how helm charts are being handled currently in my view -- at least from a best practices perspective. 

To be clear, I do not have strong opinions here. I'm just as happy to remove the ingress-auth and deal with that on my own installation. That is how we deal with protecting our kubernetes-dashboard deployment, for example. I do not see a clean way to put the ingress into the oauth-proxy chart. 

I added the ingress-auth to the PR as I do not see many use cases for deploying kibana where it would not be protected by some level of authentication. In distinction to, e.g., grafana which has in-built authentication, kibana does not have authentication inbuilt.

The ingress is built to not depend strongly on the oauth-proxy chart itself; any ingress authorization would likely work (although I have not tested it as such). 

This said, I'll happily defer to @unguiculus or any other maintainer as to how to deal with this inter-chart default challenge. If someone just let's me know a preferred default position here I will happily adapt the PR so this can get merged.

**Comment by Morriz** [comment]
Yes, leaving any (auth) implementation to the end user is preferred.

Having said all that I think removing that auth-proxy ingress and supplying it as a seperate chart (which you already did, no?) would make the merge a no brainer.

**Comment by compleatang** [comment]
Thanks to @jstriebel this should be good to go. @unguiculus hopefully everything is addressed now! :)

File: incubator/kibana/Chart.yaml
Comment (unguiculus) [inline incubator/kibana/Chart.yaml]: Can you add an icon?

**Review by unguiculus — CHANGES_REQUESTED** [review]
Looks pretty good. If you like, I'd suggest you move it to stable.

File: incubator/kibana/Chart.yaml
Comment (compleatang) [reply incubator/kibana/Chart.yaml]: 👍

**Comment by compleatang** [comment]
Have changed to stable and added the icon. Hopefully g2g now! :)

**Comment by unguiculus** [comment]
@compleatang After merging I noticed that the readme still refers to incubator. Could you create a new PR to fix this?

**Comment by jstriebel** [comment]
@unguiculus, @compleatang Thanks for the new chart! I just opened PR #2947 for the README fix.

**Comment by compleatang** [comment]
🎉  Thanks all! Especially @jstriebel @unguiculus and @Morriz

**Pull request merged**
